File: load_data.R
library(dplyr)
full_data <- read.csv("data/data_fourchoices.csv")

# mapping of button names to choice numbers
button_mapping <- c(
    "antifragile" = 1,
    "robuste" = 2,
    "fragile" = 3,
    "vulnerable" = 4
)

data <- full_data[, c("participant", "click_number", "button_name", "button_value")] %>%
    rename(participant_id = participant, trial = click_number, choice = button_name, reward = button_value) %>%
    mutate(
        option = choice,
        choice = button_mapping[choice]
    ) %>%
    select(participant_id, trial, choice, reward, option) -> data

write.csv(data, file = "data/prepared_data.csv", row.names = FALSE)

File: load_data.py
# %%
import pandas as pd

full_data = pd.read_csv("data/data_fourchoices.csv")

# %%
all_participant_data = full_data[["participant", "button_name", "button_value"]]
# %%
dict_mapping_button_name_to_index = {
    "antifragile": 0,
    "fragile": 1,
    "robuste": 2,
    "vulnerable": 3
    }
all_participant_data["choice"] = all_participant_data["button_name"].map(dict_mapping_button_name_to_index)
all_participant_data = all_participant_data.rename(columns={"button_value": "reward"})

all_participant_data["rescaled_reward"] = (all_participant_data["reward"]) / 3000 # rescale rewards to [-1,1]
all_participant_data["rescaled_reward"] = all_participant_data["rescaled_reward"].round(4)

unique_participants = all_participant_data["participant"].unique()
# %%
